users/service.go

```diff
@@ -259,16 +267,16 @@ func (svc service) UpdateClientIdentity(ctx context.Context, token, clientID, id
 
 func (svc service) GenerateResetToken(ctx context.Context, email, host string) error {
 	client, err := svc.clients.RetrieveByIdentity(ctx, email)
-	if err != nil || client.Credentials.Identity == "" {
-		return repoerr.ErrNotFound
+	if err != nil {
+		return errors.Wrap(svcerr.ErrViewEntity, err)
 	}
```

Comment:
@rodneyosodo  @WashingtonKK @dborovcanin 

What do you think about this suggestion ? 
```suggestion
	if err != nil {
		if errors.Contains(repoerr.ErrNotFound, err) {
			return nil
		} else {
			return errors.Wrap(svcerr.ErrViewEntity, err) // it should be an internal error
		}
	}
```

Reply:
Why would we return nil when the error contains `ErrNotFound` ? Shouldn't we return `svcerr.ErrNotFound` in this case?